src/main/java/de/tum/cit/aet/artemis/programming/service/tokens/UserTokenExpiryNotificationService.java

```diff
@@ -38,27 +38,32 @@ public UserTokenExpiryNotificationService(SingleUserNotificationService singleUs
     public void sendTokenExpirationNotifications() {
         log.info("Sending Token expiration notifications to single user");
         notifyOnExpiredToken();
+        notifyUserOnUpcomingVcsAccessTokenExpiry();
     }
 
     /**
-     * Notifies the user at the day of key expiry, that the key has expired
+     * Notifies the user at the day of VCS access token expiry
      */
     public void notifyOnExpiredToken() {
-        notifyUsersForKeyExpiryWindow(now().minusDays(1), now(), (user) -> {
-            singleUserNotificationService.notifyUserAboutExpiredVcsAccessToken(user);
-            return null;
-        });
+        notifyUsersForKeyExpiryWindow(now().minusDays(1), now(), singleUserNotificationService::notifyUserAboutExpiredVcsAccessToken);
     }
 
     /**
-     * Notifies users whose SSH keys are expiring within the specified date range, with the notification specified by the
+     * Notifies the user one week before the VCS access tokens expiry
+     */
+    public void notifyUserOnUpcomingVcsAccessTokenExpiry() {
```

Comment:
why only a single user? afaiu, this method will notify all users for which that is the case, right?

Reply:
good point, that would be more consistent
```suggestion
     * Notifies the user one week before the VCS access tokens expiry
     */
    public void notifyUsersOnUpcomingVcsAccessTokenExpiry() {
```